exec/vector/src/main/codegen/templates/UnionVectorWriter.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+<@pp.dropOutputFile />
+<@pp.changeOutputFile name="/org/apache/drill/exec/vector/complex/impl/UnionVectorWriter.java" />
+<#include "/@includes/license.ftl" />
+package org.apache.drill.exec.vector.complex.impl;
+
+<#include "/@includes/vv_imports.ftl" />
+
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * ListWriter-like writer with only difference that it acts only as a factory
+ * for concrete type writers for UnionVector data vector.
+ */
+public class UnionVectorWriter extends AbstractFieldWriter {
+
+  /**
+   * Map holding lazily initialized type specific writers for the union data vector
+   */
+  final Map<MinorType, FieldWriter> typeWriters = new TypeWritersMap();
```

Comment:
Why not private?

Reply:
The ```typeWriters``` field actively used by child ```UnionVectorListWriter``` class in the same package.

Reply:
Ok, thanks.